src/App.tsx

```diff
@@ -1,61 +1,114 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+
+  const newTodos = () => {
+    const newTodo = {
+      id: [...todos.sort((a, b) => b.id - a.id)][0].id + 1,
+      userId: +userId,
+      title,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    todos.push(newTodo);
+  };
+
+  const hundleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.length === 0) {
+      setTitleError(true);
+    }
+
+    if (userId === 0) {
+      setTitleError(true);
+    }
+
+    if (title.length === 0 && !userId) {
+      return;
```

Comment:
Try to combine all these checks using logical operators
https://javascript.info/logical-operators